### ISSUE TITLE
Use Zephyr RTOS to create serial driver for Magnemometer sensor

### ISSUE THREAD

**Issue by Roozki** [UBC-Snowbots/RoverFlake#34]

**Comment by stephenqiao1** [comment]
Was able to detect the imu sensor with the esp32 and zephyr rtos, but still trying to figure out why it isn't able to read and process the magnetometer data

**Comment by Roozki** [comment]
nice work so far, your code looks good to me. what is it outputting?
also how well is the serial connection? your code looks much cleaner than my implementation lol

**Comment by stephenqiao1** [comment]
Im not sure what you mean by how good the serial connection is but the output is like this right now:
![Screenshot from 2023-11-18 20-29-29](https://github.com/UBC-Snowbots/RoverFlake/assets/89325334/2240f674-17a6-460e-85d4-5f3c0bcc4eff)

**Comment by Roozki** [comment]
yeah, i want to make some packages that test serial speed and whatnot but if you haven't noticed any problems with the serial connection (connects every time, no garbage values, sends at the rates you want it to, etc) then it should be ok. That output might be something messing up with changing data types, or the IMU is just faulty. 

at line 41 in your main: `        printk("Magnetometer X: %d.%06d, Y: %d.%06d, Z: %d.%06d\n",
`
is a little odd as you are printing floats with the %d. does %f work? like:

`        printk("Magnetometer X: %0.6f, Y: %0.6f, Z: %0.6f\n",
`
or was this on purpose to print the integer as a float?

**Comment by stephenqiao1** [comment]
i used the samples they had on the zephyr site. Im trying to debug by printing log messages from the driver source files, but for some reason I cant print the LOG_DBG() messages
